.github/workflows/build.yml

```diff
@@ -1,27 +1,39 @@
-name: CI-Lint-Build-Test
-'on':
+name: build pipeline
+
+on:
   push:
     branches:
+      - master
+      - main
       - develop
       - feature/*
+      - hotfix/*
   pull_request:
-    types:
-      - opened
-      - synchronize
-      - reopened
+    branches:
+      - master
+      - main
+      - develop
+      - 'release/*'
+
+env:
+  SHOULD_PUBLISH: ${{ github.ref  == 'refs/heads/master' ||  github.ref == 'refs/heads/develop' || startsWith(github.ref , 'refs/heads/release') }}
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   # This workflow contains a single job called "build"
   build:
     # The type of runner that the job will run on
-    name: Run Lint and Deploy
+    name: Build and Deploy
     runs-on: ubuntu-latest
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+          show-progress: false
 
       - name: Use Node.js 16
```

Comment:
Why 16? That seems a bit old... I see the comment saying it's not supported, but the linked issue (https://github.com/actions/setup-node/issues/26) was closed 2 years ago already...

Reply:
@Yann-J Updated to 18